src/veh_interact.cpp

```diff
@@ -316,30 +315,19 @@ task_reason veh_interact::cant_do (char mode)
         } );
         has_tools = true; // checked later
         break;
-    case 'f': // refill mode
-        if (!ptanks.empty()) {
-            std::vector<vehicle_part *>::iterator iter;
-            for (iter = ptanks.begin(); iter != ptanks.end(); ) {
-                if ((*iter)->hp > 0 &&
-                    g->refill_vehicle_part(*veh, *iter, true)) {
-                    pass_checks = true;
-                    iter++;
-                } else {
-                    iter = ptanks.erase(iter);
-                }
-            }
-            if(pass_checks) {
-                return CAN_DO;
-            }
-        }
 
-        // if refillable parts exist but can't be refilled for some reason.
-        if (has_ptank) {
-            return CANT_REFILL;
-        }
+    case 'f': {
+        // first find all possible ammo
+        auto opts = g->u.items_with( []( const item &e ) { return e.is_ammo(); } );
```

Comment:
> The contents must be counted by charges and therefore either  AMMO  or  COMESTIBLE .

Then why are you testing for it being ammo? If you need the count-by-charges property, test for it. There are suitable static and non-static functions in the item class.

> Additionally I'm not sure what corner cases exist for non-ammo contents of fuel tanks. It possibly works but it probably doesn't and in the absence of any actual uses it's safer to document it as a precondition via the consistency check and then later relax that if necessary rather than the opposite.

You're picking one property (being ammo) of the currently used fuel types and make it mandatory. But what about the other properties? Are you sure there are no corner cases that require some other property as well?

If you're worried about some corner cases, restrict the fuel types to only those that are known to be supported.

> Your suggestion enables support for something that we don't make any use of either in mainline content or indeed any mods.

I *keeps* support for that. It does not enable it, it's already there.